docs/performance-testing.md

```diff
@@ -0,0 +1,89 @@
+
+There are 3 steps involved in doing a performance test:
+
+1. Setting up the data: This is done by executing the relevant Java script in the `setup` package (e.g. `SetupStudentProfileTest`).
+    Executing the scrip will:
+    1. Create the relevant TEAMMATES JSON data for the datastore. It is stored as `src/jmeter/resources/data/testName.json`.
+    1. Create the relevant CSV Config data for the JMeter test plan. It is stored as `src/jmeter/resources/data/testNameConfig.csv`.
+    1. Create the entities in the datastore using the JSON data. This can be verified by checking your [local datastore](http://localhost:8080/_ah/admin/datastore).
+        > The backend server (`localhost:8080`) must be started for this to happen successfully.
+1. [Run the performance test](#running-performance-tests) specified in the JMeter `.jmx` file in the `src/jmeter/tests/` directory.
+1. Remove the data that was added to the datastore (optional, but recommended). To do so, execute the relevant Java script in the `teardown` package (e.g. `DeleteStudentProfileTestData`).
+   In addition, you can choose to delete the `.json` and `.csv` files that were generated in step 1. 
+
+## Running Performance Tests
+
+Start the backend server (`localhost:8080`) before executing the JMeter performance tests.
+
+### Using Gradle
+
+To run the JMeter tests, execute this command from the main project directory:
+```
+./gradlew clean jmRun
+```
+
+If it runs without any errors:
+- A log file is generated as `build/jmeter/jmeter.log`.
+- The JMeter test results are stored as XML files with the same name as the test file `build/jmeter-report/TEST_FILE_NAME.jmx-TIMESTAMP.xml`. 
+
+> Note: If you want to retain the result files, do not include the `clean` task when running the command.
+<br/>
+
+If you want to execute specific test file(s):
+- Navigate to the `// PERFORMANCE TEST TASKS` section in `build.gradle`. 
+- Under the `jmeter` configuration:
+  - Uncomment the `jmTestFiles` parameter.
+  - Replace the dummy values with the path to the test file(s) that you want to run.
+- Then, follow the same steps mentioned above.
+
+### Using the Command Line
+
+If you have JMeter installed, you can use `jmeter` in the command line to run tests and generate reports.
+
+Execute this command after replacing the parameters with paths to the relevant files:
+```
+jmeter -n -t PATH_TO_TEST_FILE.jmx -l RESULT_FILE.csv -j LOG_FILE.log
+```
+
+### Using JMeter GUI
+
+You can also use the JMeter GUI (`./gradlew jmGui`) to run performance tests. This is particularly useful when debugging or validating that the test does what is expected.
+However, you should not use the GUI to run large scale tests as it is very resource intensive.
+
+> Remember to **disable or remove all `listeners`** if you enabled them while using the GUI since they can have a negative impact on the performance of the test.
```

Comment:
this works! work on the grammar a bit.
Why does the Gui redirects directly to the studentProfile.xml test?  Shouldn't we be able to specify the test to run using jmGui? Or it is to be build in the future?

Reply:
> work on the grammar a bit.

Not sure what you're referring to. I'll split the sentence into 2.

> Why does the Gui redirects directly to the studentProfile.xml test? Shouldn't we be able to specify the test to run using jmGui? Or it is to be build in the future?

 Similar to `jmRun`, it's linked to the `testFileDir` and `jmTestFiles` property in `build.gradle`. It opens the first available file.